pytorch_lightning/loggers/base.py

```diff
@@ -71,6 +75,15 @@ def __init__(
         self._agg_key_funcs = agg_key_funcs if agg_key_funcs else {}
         self._agg_default_func = agg_default_func
 
+    def connect(self, trainer: 'Trainer') -> None:
```

Comment:
Can you add the typing here to a proxy? I am not sure if this really works with typechecks against trainer otherwise. Or we use cast right after we created the proxy in the connector.